tinygrad/helpers.py

```diff
@@ -173,7 +173,7 @@ def fields() -> Dict[str, DType]: return DTYPES_DICT
 
   # NOTE: bfloat16 isn't supported in numpy
   # it has higher priority than float16, so least_upper_dtype(dtypes.int64, dtypes.uint64) = dtypes.float16
-  bfloat16: Final[DType] = DType(10, 2, "__bf16", None)
+  bfloat16: Final[DType] = DType(10, 2, "__nv_bfloat16", None) if os.getenv("CUDA") else DType(10, 2, "__bf16", None)
```

Comment:
wanna use `type_map` for CUDALanguage here?

Reply:
I think the right way is probably to define them in `CUDALanguage`, 'METALLanguage`, and `HIPLanguage` etc. And here we just pass like `None` so when render it uses those definitions instead.

Reply:
Yea you absolutely can't use the env var like this. What if there's two devices? What if CUDA=0?